packages/flame/lib/src/game/flame_game.dart

```diff
@@ -86,6 +86,24 @@ class FlameGame extends ComponentTreeRoot
     setMounted();
   }
 
+  /// Setting to true will remove all children and process any remaining life
+  /// cycle events.
+  bool disposeAll = false;
+
+  /// Called when the game is about to be removed from the Flutter widget tree,
+  /// but before it is actually removed.
+  ///
+  /// You can override this for additional behavior, but for [disposeAll] to be
+  /// triggered, you must call super.
+  @override
+  void onRemove() {
+    if (disposeAll) {
+      removeAll(children);
+      processLifecycleEvents();
+    }
+    super.onRemove();
+  }
```

Comment:
I was wondering if this behavior can be moved to `Component` and `ComponentTreeRoot`? Basically one handles removal of children and the other handle lifecycle events.

```dart
class Component {
  bool disposeChildrenOnRemove = false;
  void onRemove() {
    if (disposeChildrenOnRemove) {
      removeAll(children);
    }
  }
}

class ComponentTreeRoot extends Component {
  @override
  void onRemove() {
    super.onRemove();
    if (disposeChildrenOnRemove) {
      processLifecycleEvents();
    }
  }
}
```

Reply:
I think that it probably would be better to keep disposal at a top level for now and empty the global caches in there. Possibly create a new lifecycle event which is `onDispose` for the components, but I wouldn't go that far already until we really see a need for it. I don't think disposal should be this tightly coupled with `onRemove`.

Reply:
No no.  You are falling in the trap that I got into in the beginning with this.  `FlameGame` likes to think its a component (my chance at being funny), but its really like a quasi component because it isn't treated like other components.  Its the root of all components and as such, its lifecycle events are very different from other components.

Components already have `onMount`,`onRemove` and I would caution against using the term dispose in the component because, as stated above, that should be permanent and never coming back, so the user would have to re-instantiate the component.  Of which, I think that is far worse to do to a user.